### ISSUE TITLE
In hla.dat the value of codon_start in hla.dat does not fgive bp offset of first codon in translation.

### ISSUE THREAD

**Issue by colinhercus** [ANHIG/IMGTHLA#245]
Hi,

I'm using hla.dat to look for non-synonomous variants in HLA sequencing and I'm having trouble deciphering the read frame for HLA entries with partial sequence.

Here's a couple of condensed examples where CDS codon_start attribute doesn't correspond to the actual start based on the translation and DNA sequence

Example 1. codon-start=2, first codon at 3
```
FT   CDS             join(<1..270,271..546)
FT                   /codon_start=2
FT                   /product="MHC Class I HLA-A*01:01:04 sequence"
FT                   /translation="SHSMRYFFTSVSRPGRGEPRFIAVGYVDDTQFVRFDSDAASQKME
FT   exon            1..270
FT                   /number="2"
FT   exon            271..546
FT                   /number="3"
SQ   Sequence 546 BP; 112 A; 163 C; 195 G; 76 T; 0 other;
     gctcccactc catgaggtat ttcttcacat ccgtgtcccg gcccggccgc ggggagcccc        60
       S  H  S
```
Example 2. codon_start=3, actual translation starts at 2
```

FT   CDS             join(<1..44,45..314,315..590,591..866,867..983)
FT                   /codon_start=3
FT                   /allele="HLA-A*01:107"
FT                   /translation="LLLSGALALTQTWAGSHSMRYFFTSVSRPGRGEPRFIAVGYVDDT
FT                   KPLTLRWELSSQPTIPIVGIIAGLVLLGAVITGAVVAAVMWRRKSS"
FT   exon            1..44
FT                   /number="1"
FT                   /partial
SQ   Sequence 983 BP; 193 A; 303 C; 334 G; 153 T; 0 other;
     cctgctactc tcgggggccc tggccctgac ccagacctgg gcgggctccc actccatgag        60
      L  L  L   S
```
What am I doing wrong?

Thanks, Colin

**Comment by jrob119** [comment]
There are a large number of alleles, for which we have received partial sequence. For many of these sequences the start of the nucleotide sequence is part way through a codon. For this reason the reading frame will not be consistent across all sequences as they start at different points. As we have hundreds or thousands of sequences per gene, with many full-length we are able to align the sequences and determine the correct reading frame. I would also recommend reviewing our XML format, which does contain reading frame information.

**Comment by colinhercus** [comment]
I sit okay to ask what the codon_start attribute in the CDS section of
hla.dat means?

**Comment by jrob119** [comment]
Please see https://github.com/ANHIG/IMGTHLA/blob/Latest/Manual.md, for full documentation on the format;

codon_start - Indicates the offset at which the first complete codon of a coding feature can be found, relative to the first base of that feature.

**Comment by colinhercus** [comment]
Then shouldn't it be the offset in the SQ sequence for the first codon?

**Comment by colinhercus** [comment]
I did read the documentation

**Comment by jrob119** [comment]
It is the first base of the CDS sequence which has been built from the coding features which is to be used for the translation. For for some alleles we will have genomic sequence and so the sequence will contain intron or UTR, and for others we will have partial sequences where the sequence starts part way through a sequence. The number should be used as a guide to the position in the CDS (whether full or partial) to start the reading frame from. We do not use the reading frame as many alleles are partial and the start of the sequence is often an incomplete codon.  I will look at amending the documentation, to better reflect this.

**Comment by colinhercus** [comment]
Hi James,

You're basically saying what I expected it to mean but in the two examples
I gave which are partial sequences the translation for the CDS sequence
doesn't start at codon_start in the SQ sequence.

For full sequences with UTR and EXons the codon_offset is 1 in the cases I
checked. That's what I expect and I think fits your explanation.

The problem is the partial sequence where you have given a DNA sequence
that the codon_start isn't the  start of translation of the DNA sequence
for the CDS sequence. See the examples I posted.

Tomorrow I'll check the XML but I expect it will have the same problem
